ivy/functional/ivy/manipulation.py

```diff
@@ -365,15 +370,75 @@ def reshape(
     ret
         Reshaped array.
 
-    Examples
-    --------
+    This function conforms to the `Array API Standard
+    <https://data-apis.org/array-api/latest/>`_. This docstring is an extension of the
+    `docstring <https://data-apis.org/array-api/latest/API_specification/generated/signatures.manipulation_functions.reshape.html>`_ # noqa
+    in the standard. The descriptions above assume an array input for simplicity, but
+    the method also accepts :code:`ivy.Container` instances in place of
+    :code:`ivy.Array` or :code:`ivy.NativeArray` instances, as shown in the type hints
+    and also the examples below.
+
+    Functional Examples
+    -------------------
+    With :code:`ivy.Array` input:
+
     >>> x = ivy.array([[1,2,3], [4,5,6]])
     >>> y = ivy.reshape(x, (3,2))
     >>> print(y)
     ivy.array([[1, 2],
                [3, 4],
                [5, 6]])
 
+    With :code:`ivy.NativeArray` input:
+
+    >>> x = ivy.native_array([[0, 1, 2, 3]])
+    >>> y = ivy.reshape(x, (2, 2))
+    >>> print(y)
+    ivy.array([[0, 1],
+               [2, 3]])
+
+    With :code:`ivy.Container` input:
+
+    >>> x = ivy.Container(a=ivy.array([[0., 1., 2.]]), b=ivy.array([[3., 4., 5.]]))
+    >>> y = ivy.reshape(x, (-1, 1))
+    >>> print(y)
+    {
+        a: ivy.array([[0.],
+                      [1.],
+                      [2.]]),
+        b: ivy.array([[3.],
+                      [4.],
+                      [5.]])
+    }
+
+
+    Instance Method Examples
+    ------------------------
+
+    Using :code:`ivy.Array` instance method:
+
+    >>> x = ivy.array([[0., 1., 2.]])
+    >>> y = x.reshape(3, 1)
+    >>> print(y)
+    ivy.array([[0.],
+               [1.],
+               [2.]])
+
+    Using :code:`ivy.Container` instance method:
+
+    >>> x = ivy.Container(a=ivy.array([[0., 1., 2.], [3., 4., 5.]]), b=ivy.array([[6., 7., 8.], [9., 10., 11.]]))
+    >>> y = x.reshape((-1, 2))
+    >>> print(y)
+    {
+        a: ivy.array([[0., 1.],
+                    [2., 3.],
+                    [4., 5.]]),
+        b: ivy.array([[6., 7.],
+                    [8., 9.],
+                    [10., 11.]])
+    }
```

Comment:
Please add the instance method examples in the  ```array``` and ```container``` specific implementations here -:

1. ```ivy/array/manipulation.py```
2. ```ivy/container/manipulation.py```